src/main/java/org/variantsync/diffdetective/variationtree/VariationNode.java

```diff
@@ -0,0 +1,583 @@
+                    // Assert.assertTrue(ancestor.isArtifact());
+                }
+                ancestor = ancestor.getParent();
+            }
+            and.add(negate(ancestor.getDirectFeatureMapping()));
+
+            return and;
+        } else if (isArtifact()) {
+            return parent.getFeatureMappingClauses();
+        }
+
+        return List.of(getDirectFeatureMapping());
+    }
+
+    /**
+     * Returns the full feature mapping formula of this node.
+     *
+     * <p>The feature mapping of an {@link NodeType#IF} node is its
+     * {@link DiffNode#getDirectFeatureMapping direct feature mapping}. The feature mapping of
+     * {@link NodeType#ELSE} and {@link NodeType#ELIF} nodes is determined by all formulas in the
+     * respective if-elif-else chain. The feature mapping of an {@link NodeType#ARTIFACT artifact}
+     * node is the feature mapping of its parent.
+     * See Equation (1) in <a href="https://doi.org/10.1145/3540250.3549108">our paper</a>.
+     *
+     * @return the feature mapping of this node
+     */
+    public Node getFeatureMapping() {
+        final List<Node> fmClauses = getFeatureMappingClauses();
+        if (fmClauses.size() == 1) {
+            return fmClauses.get(0);
+        }
+        return new And(fmClauses);
+    }
+
+    /**
+     * Returns the presence condition clauses of this node.
+     *
+     * @return a list representing a conjunction (i.e., all clauses should be combined with boolean
+     * AND)
+     * @see getPresenceCondition
+     */
+    private List<Node> getPresenceConditionClauses() {
+        final var parent = getParent();
+
+        if (isElse() || isElif()) {
+            final List<Node> clauses = new ArrayList<>(getFeatureMappingClauses());
+
+            // Find corresponding if
+            var correspondingIf = parent;
+            while (!correspondingIf.isIf()) {
+                correspondingIf = correspondingIf.getParent();
+            }
+
+            // If this elif-else-chain was again nested in another annotation, add its pc.
+            final var outerNesting = correspondingIf.getParent();
+            if (outerNesting != null) {
+                clauses.addAll(outerNesting.getPresenceConditionClauses());
+            }
+
+            return clauses;
+        } else if (isArtifact()) {
+            return parent.getPresenceConditionClauses();
+        }
+
+        // this is mapping or root
+        final List<Node> clauses;
+        if (parent == null) {
+            clauses = new ArrayList<>(1);
+        } else {
+            clauses = parent.getPresenceConditionClauses();
+        }
+        clauses.add(getDirectFeatureMapping());
+        return clauses;
+    }
+
+    /**
+     * Returns the presence condition of this node.
+     * See Equation (2) in <a href="https://doi.org/10.1145/3540250.3549108">our paper</a>.
+     */
+    public Node getPresenceCondition() {
+        return new And(getPresenceConditionClauses());
```

Comment:
Apply same optimization as for `getFeatureMapping()`? If the number if clauses is 1, we can just return that single clause without wrapping it within an And node.

Reply:
Done.